### DISCUSSION TITLE
Data Loader Failes to Encrypt a password containsing "&".

### DISCUSSION THREAD

**Discussion by zongqua** [forcedotcom/dataloader#1006] [Q&A] [answered]
Hi,

Data Loader fails to encrypt a password and security token containing an "&". Has anyone encountered this issue and have a solution for it?

Here is any example: fjjkadfdkadf&dkjfaldfjdkfdfadfdf

The error or the output string of encryption is:
eb34e7d085604cab35630cc1bcd3baed65c9ae7715a0e9c1441a002b071e12fa
'dkjfaldfjdkfdfadfdf' is not recognized as an internal or external command,
operable program or batch file.

**Comment by ashitsalesforce** [comment, marked as answer]
Hi @zongqua , provide the command you executed and operating system on which you executed it for further troubleshooting. Here is the command I tried on MacOS after going to data loader installation folder in a terminal shell window:
```
java -cp dataloader-60.0.2.jar com.salesforce.dataloader.process.DataLoaderRunner run.mode=encrypt -e "fjjkadfdkadf&dkjfaldfjdkfdfadfdf"
```

The output was:
```
The output string of encryption is: 
15fbc1aa14b764ce71019d38e93928f318cb246e0490057f0fb3a93a0ec3eb22dcc67ff5138046f1e1feb8ed2e43e13fbca8003d365bcf6a18efec0a00d68fa3
```

I copied that and ran:
```
java -cp dataloader-60.0.2.jar com.salesforce.dataloader.process.DataLoaderRunner run.mode=encrypt -d 15fbc1aa14b764ce71019d38e93928f318cb246e0490057f0fb3a93a0ec3eb22dcc67ff5138046f1e1feb8ed2e43e13fbca8003d365bcf6a18efec0a00d68fa3
```

It resulted in the output:
```
The output string of decryption is: 
fjjkadfdkadf&dkjfaldfjdkfdfadfdf
```

**Reply by zongqua** [reply]
On a Windows PC, I used the encrypt.bat to encrypt the password and security token string. Maybe the issue is the encrypt.bat. Let me try encrypt it using Java, using the example you gave.

**Comment by zongqua** [comment]
@ashitsalesforce Just want to say thanks for the suggestion. That solves my problem.

**Comment by ashitsalesforce** [comment]
Hi @zongqua , here is more convenient workaround using `encrypt.bat` command - use double-quotes to escape & character in the password:
```
encrypt -e "fjjkadfdkadf&dkjfaldfjdkfdfadfdf"
Data Loader requires Java JRE 17 or later. Checking if it is installed...
The output string of encryption is:
043e7c9172940b1eacd49cc376c5e163fe75325cf24000dc4ac5afb2690321ca7f4fed67d3984ebb3b7c4f9dd0937431fb798ad39d17e07934b86570db5a6dcf
```
